src/common/utils/ldap/ldap.go

```diff
@@ -44,16 +44,18 @@ type Session struct {
 }
 
 // LoadSystemLdapConfig - load LDAP configure from adminserver
-func LoadSystemLdapConfig() (*Session, error) {
-
-	authMode, err := config.AuthMode()
-	if err != nil {
-		log.Errorf("can't load auth mode from system, error: %v", err)
-		return nil, err
-	}
+func LoadSystemLdapConfig(ping ...bool) (*Session, error) {
```

Comment:
Firstly the variable name is confusing, it 'd better be `skipModeCheck`, otherwise, what if we later decide we need to check the auth_mode when user triggers `ping`?

Additionally,  I briefly checked the reference of this func, I don't quite get why we need to check it at all, this func is for loading the LDAP config, the mode should not matter.

Reply:
We need to keep the auth_mode check because if current mode is not ldap_auth, the /api/ldap/users/search, /api/ldap/users/import should failed.

Reply:
Shouldn't it fail at the API level instead of `LoadLDAPConfig`?